app/controllers/articles_controller.rb

```diff
@@ -268,6 +267,15 @@ def article_params_json
     params.require(:article).permit(allowed_params)
   end
 
+  def handle_notifications(updated)
+    if updated && @article.published && @article.saved_changes["published"] == [false, true]
+      Notification.send_to_followers(@article, "Published")
+    elsif @article.saved_changes["published"] == [true, false]
+      Notification.remove_all_without_delay(notifiable_id: @article.id, notifiable_type: "Article", action: "Published")
```

Comment:
It's not clear to me why removing "Published" notifications is synchronous and removing "Reaction" notifications is async :thinking:

Reply:
The publish notifications contain the data of the article, and is sent to the followers. IMO those should be removed immediately since that data is revoked. The reaction notifications only belong to the author so I think it's okay of those are removed async.

I think based on how we handle reaction notifications though there's only one reaction notification for the article. So performance-wise it probably doesn't matter all that much.

Reply:
@lightalloy The publish notifications contain the data of the article, and is sent to the followers. IMO those should be removed immediately since that data is revoked. The reaction notifications only belong to the author so I think it's okay of those are removed async.

I think based on how we handle reaction notifications though there's only one reaction notification for the article. So performance-wise it probably doesn't matter all that much.